components/support/locale/src/main/java/mozilla/components/support/locale/LocaleManager.kt

```diff
@@ -48,6 +48,14 @@ object LocaleManager {
         return currentLocale
     }
 
+    /**
+     * Remove the locale code from the storage. Useful when having the option for a default (use
+     * system locale), in which case [getCurrentLocale] will return null
+     */
+    fun clear(context: Context) {
```

Comment:
Maybe instead of making the `clear` function public, we could rename it to something more meaningful, something like`resetToSystemDefault` that will better inform the consumer what is happening. What do you think? :)

Reply:
Your proposed name is actually much better, thanks! However, it got me thinking that actually the whole resetting should be done in this class (previously, I did it in an extension file for LocaleManager in Fenix), therefore I moved back here some logic. 

One issue I have with this is that while on Fenix I could easily test this with Mockk, on AC it seems we don't have Mockk or PowerMock, so I don't see a simple way of mocking Resources.getSystem() static method. If you have any ideas regarding this please let me know!:)

Reply:
An alternative could passing the context to `getSystemDefault` and this way you could mock the context functionality, using something like       
`ConfigurationCompat.getLocales(testContext.resources.configuration)`. What do you think?

Reply:
It's important to note that the LocaleAwareApplication overrides the app's resources at startup, so in order to get the real system chosen locale, we need to use Resources.getSystem() instead of our resources from an injected context. So I would keep the usage of these system resources inside this manager, instead of relying on the client to pass it, since it's independent of the app itself

Reply:
👍